azure/services/aso/aso_test.go

```diff
@@ -332,18 +331,17 @@ func TestCreateOrUpdateResource(t *testing.T) {
 		c := fakeclient.NewClientBuilder().
 			WithScheme(sch).
 			Build()
-		s := New(c, clusterName)
+		s := New[*asoresourcesv1.ResourceGroup](c, clusterName)
 
 		mockCtrl := gomock.NewController(t)
-		specMock := mock_azure.NewMockASOResourceSpecGetter(mockCtrl)
+		specMock := mock_azure.NewMockASOResourceSpecGetter[*asoresourcesv1.ResourceGroup](mockCtrl)
 		specMock.EXPECT().ResourceRef().Return(&asoresourcesv1.ResourceGroup{
 			ObjectMeta: metav1.ObjectMeta{
 				Name:      "name",
 				Namespace: "namespace",
 			},
 		})
-		specMock.EXPECT().Parameters(gomockinternal.AContext(), gomock.Not(gomock.Nil())).DoAndReturn(func(_ context.Context, object genruntime.MetaObject) (genruntime.MetaObject, error) {
-			group := object.DeepCopyObject().(*asoresourcesv1.ResourceGroup)
+		specMock.EXPECT().Parameters(gomockinternal.AContext(), gomock.Not(gomock.Nil())).DoAndReturn(func(_ context.Context, group *asoresourcesv1.ResourceGroup) (*asoresourcesv1.ResourceGroup, error) {
 			group.Spec.Location = ptr.To("location")
 			return group, nil
 		})
```

Comment:
This is representative of what an actual `Parameters` method implementation might look like for a service modifying an existing object.

(I think the DeepCopyObject that used to exist was redundant even before these changes.)